homeassistant/components/google/__init__.py

```diff
@@ -247,27 +249,25 @@ def setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     token_file = hass.config.path(TOKEN_FILE)
     if not os.path.isfile(token_file):
+        _LOGGER.debug("Token file does not exist, authenticating for first time")
         do_authentication(hass, config, conf)
     else:
-        if not check_correct_scopes(token_file, conf):
+        if not check_correct_scopes(hass, token_file, conf):
+            _LOGGER.debug("Existing scopes are not sufficient, re-authenticating")
             do_authentication(hass, config, conf)
         else:
             do_setup(hass, config, conf)
 
     return True
 
 
-def check_correct_scopes(token_file, config):
+def check_correct_scopes(hass, token_file, config):
     """Check for the correct scopes in file."""
-    with open(token_file, encoding="utf8") as tokenfile:
-        contents = tokenfile.read()
-
-        # Check for quoted scope as our scopes can be subsets of other scopes
-        target_scope = f'"{config.get(CONF_CALENDAR_ACCESS).scope}"'
-        if target_scope not in contents:
-            _LOGGER.warning("Please re-authenticate with Google")
-            return False
-    return True
+    creds = Storage(hass.config.path(TOKEN_FILE)).get()
+    if not creds or not creds.scopes:
+        return False
+    target_scope = config.get(CONF_CALENDAR_ACCESS).scope
```

Comment:
There's a default for this config option in the config schema.
```suggestion
    target_scope = config[CONF_CALENDAR_ACCESS].scope
```